src/libraries/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.SymbolicLink.cs

```diff
@@ -0,0 +1,129 @@
+
+            // Act - Assert
+            Assert.Throws<FileNotFoundException>(() => watcher.EnableRaisingEvents = true);
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void FileSystemWatcher_DirectorySymbolicLink_TargetsSelf_Fails()
+        {
+            // Arrange
+            string linkPath = GetTestFilePath();
+            CreateSymbolicLinkToTarget(targetPath: linkPath, isDirectory: true, linkPath: linkPath);
+            using var watcher = new FileSystemWatcher(linkPath);
+
+            // Act - Assert
+            Assert.Throws<FileNotFoundException>(() => watcher.EnableRaisingEvents = true);
+        }
+
+        [Fact]
+        public void FileSystemWatcher_SymbolicLink_TargetsDirectory_Create()
+        {
+            // Arrange
+            using var tempDir = new TempDirectory(GetTestFilePath());
+            string linkPath = CreateSymbolicLinkToTarget(tempDir.Path, isDirectory: true);
+
+            using var watcher = new FileSystemWatcher(linkPath);
+            watcher.NotifyFilter = NotifyFilters.DirectoryName;
+
+            string subDirName = GetTestFileName();
+            string subDirPath = Path.Combine(tempDir.Path, subDirName);
+
+            // Act - Assert
+            ExpectEvent(watcher, WatcherChangeTypes.Created,
+                action: () => Directory.CreateDirectory(subDirPath),
+                cleanup: () => Directory.Delete(subDirPath),
+                expectedPath: Path.Combine(linkPath, subDirName));
+        }
+
+        [Fact]
+        public void FileSystemWatcher_SymbolicLink_TargetsDirectory_Create_IncludeSubdirectories()
+        {
+            // Arrange
+            const string subDir = "subDir";
+            const string subDirLv2 = "subDirLv2";
+            using var tempDir = new TempDirectory(GetTestFilePath());
+            using var tempSubDir = new TempDirectory(Path.Combine(tempDir.Path, subDir));
+
+            string linkPath = CreateSymbolicLinkToTarget(tempDir.Path, isDirectory: true);
+            using var watcher = new FileSystemWatcher(linkPath);
+            watcher.NotifyFilter = NotifyFilters.DirectoryName;
+
+            string subDirLv2Path = Path.Combine(tempSubDir.Path, subDirLv2);
+
+            // Act - Assert
+            ExpectNoEvent(watcher, WatcherChangeTypes.Created,
+                action: () => Directory.CreateDirectory(subDirLv2Path),
+                cleanup: () => Directory.Delete(subDirLv2Path));
+
+            // Turn include subdirectories on.
+            watcher.IncludeSubdirectories = true;
+
+            ExpectEvent(watcher, WatcherChangeTypes.Created,
+                action: () => Directory.CreateDirectory(subDirLv2Path),
+                cleanup: () => Directory.Delete(subDirLv2Path),
+                expectedPath: Path.Combine(linkPath, subDir, subDirLv2));
+        }
+
+        [Fact]
+        public void FileSystemWatcher_SymbolicLink_IncludeSubdirectories_DoNotDereferencesChildLink()
```

Comment:
So we expect no events here because we don't watch subfolder symbolic links, correct? The only symlink we watch is the root folder, if it happens to be one.

nit: Typo

```suggestion
        public void FileSystemWatcher_SymbolicLink_IncludeSubdirectories_DoNotDereferenceChildLink()
```

Reply:
> So we expect no events here because we don't watch subfolder symbolic links, correct?

Yup.